src/home_robot_spot/home_robot_spot/spot_client.py

```diff
@@ -0,0 +1,417 @@
+    ),
+    ("frontright_fisheye_image", RGB_FORMAT, RGB_THRESH, cv2.ROTATE_90_CLOCKWISE),
+    ("left_depth_in_visual_frame", DEPTH_FORMAT, BODY_THERSH, None),
+    ("left_fisheye_image", RGB_FORMAT, RGB_THRESH, None),
+    ("right_depth_in_visual_frame", DEPTH_FORMAT, BODY_THERSH, cv2.ROTATE_180),
+    ("right_fisheye_image", RGB_FORMAT, RGB_THRESH, cv2.ROTATE_180),
+]
+
+SENSOR_FRAME_NAMES = [
+    "hand_color_image_sensor",
+    "hand_color_image_sensor",
+    "back_fisheye",
+    "back_fisheye",
+    "frontleft_fisheye",
+    "frontleft_fisheye",
+    "frontright_fisheye",
+    "frontright_fisheye",
+    "left_fisheye",
+    "left_fisheye",
+    "right_fisheye",
+    "right_fisheye",
+]
+
+
+class SpotPublishers:
+    def __init__(self, spot: Spot, rotate=False, quality_percent=75, verbose=False):
+        self.spot = spot
+        self.sources = CAMERA_SOURCES
+        self.reqs = [
+            build_image_request(
+                name,
+                quality_percent=quality_percent,
+                pixel_format=format,
+                resize_ratio=1,
+            )
+            for name, format, _, _ in self.sources
+        ]
+        self.rotate = rotate
+        self.finished_initial_capture = False
+        self.verbose = verbose
+
+    def start(self):
+        # Should we do one thread per observation? It currently takes
+        self.threads = [
+            Thread(target=self.update_obs),
+            Thread(target=self.update_obstacle_map),
+        ]
+        for thread in self.threads:
+            thread.daemon = True
+            thread.start()
+
+        self.updated = False
+
+        print("waiting for first observation")
+        while not self.updated:
+            time.sleep(0.1)
+        print("got first observation")
+
+    def stop(self):
+        # Close all threads
+        for thread in self.threads:
+            thread.join()
+
+    def update_obs(
+        self,
+    ):
+        last_update = time.time()
+        while True:
+            self.observations = self.get_cam_observations()
+            self.observations.update(self.get_obstacle_map())
+            self.updated = True
+            if self.verbose:
+                print("FPS: ", 1 / (time.time() - last_update))
+            last_update = time.time()
+
+    def update_obstacle_map(self):
+        while True:
+            self.observations.update(self.get_obstacle_map())
+
+    def rot2npy(self, rotation):
+        return np.array(
+            [
+                rotation.w,
+                rotation.x,
+                rotation.y,
+                rotation.z,
+            ]
+        )
+
+    def intrins_to_dict(self, intrins):
+        return dict(
+            fx=intrins.focal_length.x,
+            fy=intrins.focal_length.y,
+            px=intrins.principal_point.x,
+            py=intrins.principal_point.y,
+        )
+
+    def get_cam_observations(self):
+        responses = self.spot.image_client.get_image(self.reqs)
```

Comment:
We can maybe split this function so that it takes in a set of responses - that way we can save the responses separately